### ISSUE TITLE
[OpenBMC] seeing strange MTM and SN values from OpenBMC

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3993]
Something new with the VPD is causing this to appear from the REST API. .

```
[root@briggs01 ~]# bmcdiscover --range 172.12.253.100-176
172.12.253.168,8335-GTC........,1318C4A.........,,,mp,bmc,,
[root@briggs01 ~]# bmcdiscover --range 172.12.253.100-176 -w
Writing node-8335-gtc---------1318c4a--------- (172.12.253.168,8335-GTC........,1318C4A.........,,,mp,bmc,,) to database...
[root@briggs01 ~]# 
```

We need to fix 
* dodiscovery
* bmcdiscover

to remove those dots and follow up with FW team.

**Comment by zet809** [comment]
With fix in #3994 , the output looks like this, so I close this issue. Thx @whowutwut and @xuweibj 
```
# bmcdiscover --range  172.12.139.105
172.12.139.105,8335-GTC,1318C4A,,,mp,bmc,,
# lsdef mid05tor12cn05 -i mtm,serial
Object name: mid05tor12cn05
    mtm=8335-GTC
    serial=1318C4A
```

**Comment by whowutwut** [comment]
This is good for xCAT, but petitboot still shows... 
```
 Petitboot (v1.5.1-p36336db)                  8335-GTC........ 1318C4A.........
 ──────────────────────────────────────────────────────────────────────────────
  [Disk: sda2 / d3b8847a-d933-4611-b06d-21f753633851]
    Red Hat Enterprise Linux Server (0-rescue-66afe22050bd41e187b8c8124d721158)
 *  Red Hat Enterprise Linux Server (4.11.0-30.el7a.ppc64le) 7.4 (Pegas)       
  [Network: enP5p1s0f0 / 70:e2:84:14:28:0e]
    xCAT Genesis shell

  System information
  System configuration
  System status log
  Language
  Rescan devices
  Retrieve config from URL    
  Exit to shell






 ──────────────────────────────────────────────────────────────────────────────
 Enter=accept, e=edit, n=new, x=exit, l=language, g=log, h=help```

So now we are hiding this potential problem. I wonder how it got to this in the first place...   It's not firmware related i don't think..
